core/src/main/scala/org/apache/spark/scheduler/SchedulerBackend.scala

```diff
@@ -41,4 +41,11 @@ private[spark] trait SchedulerBackend {
    */
   def applicationId(): String = appId
 
+  /**
+   * Get an application ID associated with the job.
+   *
```

Comment:
actually, does it make sense for applications running in client mode to have an attempt ID?